### PR TITLE
feat(spel): Add feature flag for DoNotEval SpEL.

### PR DESCRIPTION
**Pull request by armory-abedonik**
**Usage**

```
# orca-local.yml

expression:
  do-not-eval-spel:
    enabled: true
```

**References**:

* https://github.com/spinnaker/kork/pull/978
* https://github.com/spinnaker/orca/pull/4309

### CODE REVIEW

**Review by DanielaS12 — APPROVED** [review]
looks good to me

**Review by kkotula — APPROVED** [review]

File: kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/ExpressionsSupport.java [hunk 1]
```diff
@@ -105,6 +107,11 @@ public ExpressionsSupport(
       expressionFunctionProviders.addAll(
           pluginManager.getExtensions(ExpressionFunctionProvider.class));
     }
+
+    if (dynamicConfigService.isEnabled("expression.do-not-eval-spel.enabled", false)) {
```
Comment (dbyron-sf) [inline kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/ExpressionsSupport.java:111]: I'd prefer a "real" config class here to using the DynamicConfigService.  I think it's more discoverable that way.  As in, the properties then show up in spring boot's `/configprops` endpoint.  As well, if we end up adding more properties I think it's helpful that they stay together.

Comment (armory-abedonik) [reply kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/ExpressionsSupport.java:111]: Hi @dbyron-sf, as I can see this interface was created especially for Feature Flags. Also, this interface is used in a lot of places, so it looks like the general approach for working with Feature Flags. I'd like to ensure code consistency.

Screenshot 2022-10-13 at 18 25 52

Comment (dbyron-sf) [reply kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/ExpressionsSupport.java:111]: I suppose.  I think it's a bit of an anti-pattern though.  There are also way more instances of using spring config classes....I'm sticking to my story.

Comment (armory-abedonik) [reply kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/ExpressionsSupport.java:111]: I just checked, yes, there are a lot of places based on static configuration but it mostly feature configuration. It's not a feature flag (on/off). I'm worrying about the interface is used somehow from UI to define feature flags in the runtime...

Comment (armory-abedonik) [reply kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/ExpressionsSupport.java:111]: I'm wondering, is here any documentation for Feature Flags management? 🤔

Comment (dbyron-sf) [reply kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/ExpressionsSupport.java:111]: There's https://github.com/spinnaker/governance/pull/303/files, but it's not exactly what you're looking for.

**Comment by dbyron-sf** [comment]
@mergifyio update

File: kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/config/ExpressionProperties.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.kork.expressions.config;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@Getter
```
Comment (dbyron-sf) [inline kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/config/ExpressionProperties.java:23]: I've used [`@Data`](https://projectlombok.org/features/Data) for classes like this. See e.g. [KubernetesConfigurationProperties](https://github.com/spinnaker/clouddriver/blob/fb5c326c798dff8fe8a2b94d698555dde1da3444/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/config/KubernetesConfigurationProperties.java#L22).

Seems like you started with `@Data` and then switched.  I'm curious why.

**Review by dbyron-sf — COMMENTED** [review]
Thanks for making the changes.  One small question, and then I think we're good.

File: kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/config/ExpressionProperties.java (hunk 2 above)
Comment (armory-abedonik) [reply kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/config/ExpressionProperties.java:23]: `@Data` annotation doesn't work with the super class well. Also, the annotations besides Getter and Setter aren't required here, as a result, I took a decision to define only required stuff.

Comment (dbyron-sf) [reply kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/config/ExpressionProperties.java]: Sounds good.  Thanks.

**Review by dbyron-sf — APPROVED** [review]

**Comment by dogonthehorizon** [comment]
@mergifyio backport release-1.28.x release-1.29.x

**Comment by dogonthehorizon** [comment]
Backporting to ensure that these PRs are appropriately feature flagged:
- https://github.com/spinnaker/kork/pull/984
- https://github.com/spinnaker/kork/pull/985

**Comment by dbyron-sf** [comment]
@dogonthehorizon or @armory-abedonik , mind adding a release note to mention the new flag/feature at https://github.com/spinnaker/spinnaker.io/blob/master/content/en/docs/releases/next-release-preview/index.md?

**Comment by armory-abedonik** [comment]
@dbyron-sf , sure, it's here https://github.com/spinnaker/spinnaker.io/pull/257

**Comment by dbyron-sf** [comment]
Thanks!

**Pull request merged**